drivers/mcp2515/candev_mcp2515.c

```diff
@@ -0,0 +1,726 @@
+            _neednewisr = 1;
+            return;
+        }
+    }
+}
+
+static int _set(candev_t *candev, canopt_t opt, void *value, size_t value_len)
+{
+    candev_mcp2515_t *dev = (candev_mcp2515_t *)candev;
+    int res = 0;
+
+    DEBUG("Inside mcp2515 set opt=%d\n", opt);
+    switch (opt) {
+    case CANOPT_BITTIMING:       /**< bit timing parameter */
+        if (value_len < sizeof(candev->bittiming)) {
+            res = -EOVERFLOW;
+        }
+        else {
+            memcpy(&candev->bittiming, value, sizeof(candev->bittiming));
+            res = _init(candev);
+            if (res == 0) {
+                res = sizeof(candev->bittiming);
+            }
+
+        }
+        break;
+    case CANOPT_STATE:
+        if (value_len < sizeof(uint8_t)) {
+            res = -EOVERFLOW;
+        }
+        else {
+            switch (*((canopt_state_t *)value)) {
+            case CANOPT_STATE_LISTEN_ONLY:
+                if (mutex_trylock(&_mcp_mutex)) {
+                    res = mcp2515_set_mode(dev, MODE_LISTEN_ONLY);
+                    mutex_unlock(&_mcp_mutex);
+                }
+                else {
+                    DEBUG("set1_Failed to lock mutex\n");
+                    return -1;
+                }
+                break;
+            case CANOPT_STATE_OFF:
+            case CANOPT_STATE_SLEEP:
+                if (mutex_trylock(&_mcp_mutex)) {
+                    res = mcp2515_set_mode(dev, MODE_SLEEP);
+                    mutex_unlock(&_mcp_mutex);
+                }
+                else {
+                    DEBUG("set2_Failed to lock mutex\n");
+                    return -1;
+                }
+                break;
+            case CANOPT_STATE_ON:
+                if (mutex_trylock(&_mcp_mutex)) {
+                    res = mcp2515_set_mode(dev, MODE_NORMAL);
+                    mutex_unlock(&_mcp_mutex);
+                }
+                else {
+                    DEBUG("set3_Failed to lock mutex\n");
+                    return -1;
+                }
+                break;
+            default:
+                res = -ENOTSUP;
+                break;
+            }
+        }
+        break;
+    default:
+        res = -ENOTSUP;
+    }
+
+    return res;
+}
+
+static int _get(candev_t *candev, canopt_t opt, void *value, size_t max_len)
+{
+    candev_mcp2515_t *dev = (candev_mcp2515_t *)candev;
+    int res = 0;
+
+    DEBUG("Inside mcp2515 get opt=%d\n", opt);
+    switch (opt) {
+    case CANOPT_BITTIMING:
+        if (max_len < sizeof(candev->bittiming)) {
+            res = -EOVERFLOW;
+        }
+        else {
+            memcpy(value, &candev->bittiming, sizeof(candev->bittiming));
+            res = sizeof(candev->bittiming);
+        }
+        break;
+    case CANOPT_RX_FILTERS:      /**< rx filters */
+        if (max_len % sizeof(struct can_filter) != 0) {
+            res = -EOVERFLOW;
+        }
+        else {
+            res = 1;
+        }
```

Comment:
From the original PR I saw that there was a comment asking about the return value here, it seems no rx_filter is returned.

Reply:
Mhmmm.... this one might still be tricky. It looks like it is not implemented yet. But I'm also not sure if it is possible to implement this, because the the candev for the mcp holds different parameters than the candev for stm32.

I'll need some time to investigate whether this is possible or not, and some more time to implement it (if it is).

Reply:
Its ok for an option not to be implemented, so if it can't maybe you could return a `-ENOTSUP`?, let me know what you find out.

Reply:
I will add -ENOTSUP for now and add a comment above it with some more explanation. 

I might still add it in the future, but I currently don't have much time to spare. 

The impact of the not implemented feature should be very minor, because it isn't used anywhere significant at the moment (and the candev test-app doesn't use it at all).

Do you agree with this?

Reply:
I don't know much about CAN, so I'll go with your judgment here :)

Reply:
> I don't know much about CAN.

To gauge the impact of the missing future.